src/main/java/com/aws/greengrass/mqttbridge/clients/MQTTClient.java

```diff
@@ -218,14 +218,27 @@ private synchronized void updateSubscriptionsInternal() {
         Set<String> topicsToSubscribe = new HashSet<>(toSubscribeLocalMqttTopics);
         topicsToSubscribe.removeAll(subscribedLocalMqttTopics);
 
-        // TODO: Support configurable qos, add retry
+        // TODO: Support configurable qos
         topicsToSubscribe.forEach(s -> {
-            try {
-                mqttClientInternal.subscribe(s);
-                LOGGER.atDebug().kv(TOPIC, s).log("Subscribed to topic");
-                subscribedLocalMqttTopics.add(s);
-            } catch (MqttException e) {
-                LOGGER.atError().kv(TOPIC, s).log("Failed to subscribe");
+            int waitBeforeRetry = MIN_WAIT_RETRY_IN_SECONDS;
+            while (true) {
+                try {
```

Comment:
Check if the thread is interrupted before doing work and quit if we are interrupted.

Reply:
updated to work until uninterrupted

Reply:
updated to work until uninterrupted